go-controller/pkg/node/node.go

```diff
@@ -294,18 +294,27 @@ func (n *OvnNode) WatchEndpoints() error {
 			}
 			newEndpointAddressMap := buildEndpointAddressMap(epNew.Subsets)
 			for _, subset := range epOld.Subsets {
+				//build the array of ports
+				var ports []int32
+				for _, port := range subset.Ports {
+					ports = append(ports, port.Port)
+				}
 				for _, address := range subset.Addresses {
 					if _, ok := newEndpointAddressMap[address.IP]; !ok {
-						deleteConntrack(address.IP)
+						deleteConntrack(address.IP, ports)
```

Comment:
@JacobTanenbaum I was thinking about something like this:

```
type epAddressItem struct {
    ip string
    port int32
}

// buildUDPEndpointAddressMap builds a map of all UDP ports in the endpoint subset
// along with that port's IP address
func buildUDPEndpointAddressMap(epSubsets []kapi.EndpointSubset) map[string]struct{} {
	epMap := make(map[*epAddressItem]struct{})
	for _, subset := range epSubsets {
		for _, address := range subset.Addresses {
			for _, port := range subset.Ports {
				if port.Protocol == UDP {
					epMap[&epAddressItem{
						ip: address.IP,
						port: port.Port,
					}] = struct{}{}
				}
			}
		}
	}
	return epMap
}


	newEpAddressMap := buildUDPEndpointAddressMap(epNew.Subsets)
	for item := range buildUDPEndpointAddressMap(epOld.Subsets) {
		if _, ok := newEpAddressMap[item]; !ok {
			deleteConntrack(item.IP, item.port)
		}
	}
```

and the same for Delete... build the map, iterate, delete each IP/port on its own rather than as a port array.

Reply:
done